### ISSUE TITLE
NPE with no custom headers

### ISSUE THREAD

**Issue by jaakla** [OpenLocate/openlocate-android#85]
If no custom headers are set for endpoint, then app crashes:

Code:
```java
 endpoints.add(OpenLocate.Endpoint.builder("https://myendpoint/"+BuildConfig.UUID)
      // .withHeader("minuheader", "Header's value")
    .build());

```

Crash:
```text
05-16 11:37:35.104 23805-23805/com.openlocate.example:LocationService E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.openlocate.example:LocationService, PID: 23805
    java.lang.RuntimeException: Unable to create application com.openlocate.example.ExampleApplication: java.lang.NullPointerException: Attempt to invoke virtual method 'java.util.Set java.util.HashMap.entrySet()' on a null object reference
        at android.app.ActivityThread.handleBindApplication(ActivityThread.java:5743)
        at android.app.ActivityThread.-wrap1(Unknown Source:0)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1656)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:164)
        at android.app.ActivityThread.main(ActivityThread.java:6494)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807)
     Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'java.util.Set java.util.HashMap.entrySet()' on a null object reference
        at com.openlocate.android.core.OpenLocate$Endpoint.toJson(OpenLocate.java:326)
        at com.openlocate.android.core.OpenLocate.saveConfiguration(OpenLocate.java:657)
        at com.openlocate.android.core.OpenLocate.initialize(OpenLocate.java:468)
        at com.openlocate.example.ExampleApplication.onCreate(ExampleApplication.java:33)
        at android.app.Instrumentation.callApplicationOnCreate(Instrumentation.java:1119)
        at android.app.ActivityThread.handleBindApplication(ActivityThread.java:5740)
        at android.app.ActivityThread.-wrap1(Unknown Source:0) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1656) 
        at android.os.Handler.dispatchMessage(Handler.java:106) 
        at android.os.Looper.loop(Looper.java:164) 
        at android.app.ActivityThread.main(ActivityThread.java:6494) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807) 
```

**Comment by kurabi** [comment]
Thanks, will take a look.

**Comment by kurabi** [comment]
This is resolved as per #88
